r_style_guide.md

```diff
@@ -0,0 +1,25 @@
+# Style guide for R
+
+R is a commonly used language for statistical analysis and data visualisation, however unlike Python, conventions around styling are varied. At CPG, we follow the [Tidyverse Style Guide](https://style.tidyverse.org/); the full overview of syntax can be viewed [here](https://style.tidyverse.org/syntax.html). As laid out in the documentation,  [snake_case](https://en.wikipedia.org/wiki/Snake_case) is used for variable and file naming, with short variable names preferred over long ones (try keeping them to a maximum of two words). Additionally, each R script should be documented with a single or multi-line comment at the start of the script, explaining what the script does and how to use it.
+
+## Code editors and environments
+For those that use RStudio, [Styler](https://www.tidyverse.org/blog/2017/12/styler-1.0.0/) can be used as a source code formatter to ensure proper styling. Otherwise, [lintr](https://github.com/r-lib/lintr) can be used to confirm the style guide is adhered to and flag any inconsistencies. 
```

Comment:
I think Styler can be used with any set-up by just using the commands from the package - did we want to say that Styler reformatting is a requirement?
I'm just thinking that without it being mandatory, we would have situations where one person writes their code without it and just uses lintr. Someone then adds to the code and runs Styler, which would then change additional formatting choices throughout, eg. how Styler handles open brackets over multiple lines

Reply:
Oh interesting, I didn't know that Styler could be run outside of RStudio (I just tested it and it works well). From my understanding, lintr (when set to the default config) and Styler both follow Tidyverse, so they should be pretty similar, however as you pointed out, there might be some minor differences in indentation/handling open brackets. I guess the larger question is if we care about this? I'm happy either way, just curious what other people think.

Reply:
Yep no conflicts between their default rules, it would only clutter some occasional PR's with the line reformatting I think. So, not a critical decision. It's looking great!

Reply:
Both styler and lintr look good to me. I tested them both in Visual Studio Code and it works well. However I think we should choose one and stick to it to avoid what Michael just mentioned about reformatting. I like the Pretty Printing of Styler, so I would vote for the later.